### PR TITLE
Update the tflint template for avm 0.5.0 and add new rules

### PR DESCRIPTION
**Pull request by jaredfholgate**
Update to use the latest avm linting and add the new rules to the config.

### CODE REVIEW

File: avm.tflint.hcl [hunk 1]
```diff
@@ -136,6 +136,62 @@ rule "azurerm_resource_tag" {
...
+rule "private_endpoints" {
+  enabled = true
+}
+
+rule "private_endpoints" {
+  enabled = true
+}
+
+rule "role_assignments" {
+  enabled = true
```
Comment (matt-FFFFFF) [inline avm.tflint.hcl:192]: These rules are enabled by default, do we need them here?

**Review by matt-FFFFFF — COMMENTED** [review]
one comment otherwise lgtm

File: avm.tflint.hcl (hunk 1 above)
Comment (jaredfholgate) [reply avm.tflint.hcl:192]: I was unsure about this, but I thought surfacing them would be more user friendly for those that want to override them. It helped me to understand what rules are included in avm. But also appreciate your point. I happy either way. Can remove them if you like.

Comment (matt-FFFFFF) [reply avm.tflint.hcl]: The others are in there, so we can keep them in - all good

**Review by matt-FFFFFF — APPROVED** [review]

**Comment by matt-FFFFFF** [comment]
(ignore the build error - will be fixed by #42

**Pull request merged**